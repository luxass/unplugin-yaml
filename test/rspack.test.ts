import { readFile } from "node:fs/promises";
import { tmpdir } from "node:os";
import path from "node:path";
import { rspack as createRspack } from "@rspack/core";
import { expect, it } from "vitest";
import type { Configuration, Stats } from "@rspack/core";

import YAMLPlugin from "../src/rspack";
import { removeComments } from "./utils";

const dir = path.join(tmpdir(), "unplugin-yaml-tests");
async function rspack(config: Configuration): Promise<Stats | undefined> {
  return new Promise((resolve, reject) => {
    const compiler = createRspack({
      optimization: {
        minimize: true,
      },
      output: {
        path: dir,
        filename: `${Date.now()}-bundle.js`,
      },
      mode: "production",
      ...config,
      experiments: {
        rspackFuture: {
          // disables the bundler info
          bundlerInfo: {
            force: false,
          },
        },
      },
    });

    compiler.run((err, stats) => {
      if (err) {
        reject(err);
      }

      resolve(stats);
    });
  });
}

it("expect yaml import to be a json object", async () => {
  const result = await rspack({
    entry: "./test/fixtures/js-yaml.js",
    plugins: [
      YAMLPlugin(),
    ],
  });

  const json = result?.toJson();
  expect(json).toBeDefined();

  const file = json!.assetsByChunkName!.main;
  const content = await readFile(path.join(dir, file![0]!), "utf-8");

  expect(removeComments(content)).toMatchSnapshot();
});

it("expect yaml import to be a string", async () => {
  const result = await rspack({
    entry: "./test/fixtures/js-yaml-raw.js",
    plugins: [
      YAMLPlugin(),
    ],
  });

  const json = result?.toJson();
  expect(json).toBeDefined();

  const file = json!.assetsByChunkName!.main;
  const content = await readFile(path.join(dir, file![0]!), "utf-8");

  expect(removeComments(content)).toMatchSnapshot();
});

it("expect yml import to be a json object", async () => {
  const result = await rspack({
    entry: "./test/fixtures/js-yml.js",
    plugins: [
      YAMLPlugin(),
    ],
  });

  const json = result?.toJson();
  expect(json).toBeDefined();

  const file = json!.assetsByChunkName!.main;
  const content = await readFile(path.join(dir, file![0]!), "utf-8");

  expect(removeComments(content)).toMatchSnapshot();
});

it("expect yml import to be a string", async () => {
  const result = await rspack({
    entry: "./test/fixtures/js-yml-raw.js",
    plugins: [
      YAMLPlugin(),
    ],
  });

  const json = result?.toJson();
  expect(json).toBeDefined();

  const file = json!.assetsByChunkName!.main;
  const content = await readFile(path.join(dir, file![0]!), "utf-8");

  expect(removeComments(content)).toMatchSnapshot();
});
