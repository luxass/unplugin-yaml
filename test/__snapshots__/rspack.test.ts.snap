// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`expect yaml import to be a json object 1`] = `"(()=>{"use strict";var r={},e={};function n(t){var o=e[t];if(void 0!==o)return o.exports;var i=e[t]={exports:{}};return r[t](i,i.exports,n),i.exports}n.rv=function(){return"1.0.0-alpha.5"},n.ruid="bundler=rspack@1.0.0-alpha.5";console.log({pluginDir:"./plugins",web:{enabled:!0},logging:{type:"stdout",level:"info"}})})();"`;

exports[`expect yaml import to be a string 1`] = `"(()=>{"use strict";var n={},r={};function e(t){var o=r[t];if(void 0!==o)return o.exports;var u=r[t]={exports:{}};return n[t](u,u.exports,e),u.exports}e.rv=function(){return"1.0.0-alpha.5"},e.ruid="bundler=rspack@1.0.0-alpha.5";console.log("pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n")})();"`;

exports[`expect yml import to be a json object 1`] = `"(()=>{"use strict";var r={},e={};function n(t){var o=e[t];if(void 0!==o)return o.exports;var i=e[t]={exports:{}};return r[t](i,i.exports,n),i.exports}n.rv=function(){return"1.0.0-alpha.5"},n.ruid="bundler=rspack@1.0.0-alpha.5";console.log({pluginDir:"./plugins",web:{enabled:!0},logging:{type:"stdout",level:"info"}})})();"`;

exports[`expect yml import to be a string 1`] = `"(()=>{"use strict";var n={},r={};function e(t){var o=r[t];if(void 0!==o)return o.exports;var u=r[t]={exports:{}};return n[t](u,u.exports,e),u.exports}e.rv=function(){return"1.0.0-alpha.5"},e.ruid="bundler=rspack@1.0.0-alpha.5";console.log("pluginDir: ./plugins\\n\\nweb:\\n  enabled: true\\n\\nlogging:\\n  type: stdout\\n  level: info\\n")})();"`;
